integrations/repo_search_test.go

```diff
@@ -27,6 +31,23 @@ func resultFilenames(t testing.TB, doc *HTMLDoc) []string {
 func TestSearchRepo(t *testing.T) {
 	prepareTestEnv(t)
 
+	repo, err := models.GetRepositoryByOwnerAndName("user2", "repo1")
+	assert.NoError(t, err)
+
+	models.UpdateRepoIndexer(repo)
+
+	log.Printf("Waiting for indexing\n")
+
+	i := 0
+	for {
+		if repo.IndexerStatus != nil && len(repo.IndexerStatus.CommitSha) != 0 && i <= 60 {
```

Comment:
What happens if this takes more than 60 seconds? wouldn't it get stuck in an infinite loop?

Reply:
```suggestion
		if (repo.IndexerStatus != nil && len(repo.IndexerStatus.CommitSha) != 0) || i > 60 {
```

Oh dear what was wrong with me!

Reply:
Go test would have timed out with a failure but still...